updater/driver/v1/types.go

```diff
@@ -26,11 +26,18 @@ type EnrichmentRecord struct {
 // ParsedVulnerabilities is an entity-component system describing discovered
 // vulnerabilities.
 type ParsedVulnerabilities struct {
-	Updater       string
+	// Updater is an ID for the Updater that produced this set of
+	// Vulnerabilities.
+	Updater string
+	// Vulnerability is the set of new and modified Vulnerabilities.
 	Vulnerability []Vulnerability
-	Package       []Package
-	Distribution  []Distribution
-	Repository    []Repository
+	// The following are indexed into via Vulnerability structs.
+	Package      []Package
+	Distribution []Distribution
+	Repository   []Repository
+	// Removed is a list of Vulnerabilities (the "Name" member) removed in the
+	// current set of Vulnerabilities.
+	Removed []string
```

Comment:
Just to mention: I had the realization that for us the `Name` is generic i.e. for some VEX CVE files we generate 100s of vulnerabilities with the same name (CVE-2023-001 for samba, CVE-2023-001 for samba-libs etc). This didn't actually change much except that the `UpdateVulnerabilities()` method now parses existing vulnerabilities into a `map[string][]string`, where the key is the name (CVE-2023-001) and the value is a list of all vuln ID that share that name.

Reply:
All that to say, I think this is still valid, it's just worth pointing out the nuanced difference

Reply:
Right -- this isn't in use yet, but I imagine this is in the current Updater's namespace